pynguin log ./proposals/spec_kmeans.py
[01:50:40] INFO     Start Pynguin Test Generation…              generator.py:116
           INFO     Collecting static constants from module     generator.py:216
                    under test                                                  
           INFO     No constants found                          generator.py:219
           INFO     Setting up runtime collection of constants  generator.py:228
<unknown>:3: DeprecationWarning: invalid escape sequence '\|'
/opt/conda/envs/redistricting/lib/python3.10/inspect.py:469: DeprecationWarning: `_dispatchable._fallback_to_nx` is deprecated and will be removed in NetworkX v3.5. Use `nx.config.fallback_to_nx` instead.
  value = getattr(object, key)
<unknown>:4: DeprecationWarning: invalid escape sequence '\m'
<unknown>:5: DeprecationWarning: invalid escape sequence '\l'
<unknown>:4: DeprecationWarning: invalid escape sequence '\s'
<unknown>:5: DeprecationWarning: invalid escape sequence '\g'
<unknown>:5: DeprecationWarning: invalid escape sequence '\{'
<unknown>:5: DeprecationWarning: invalid escape sequence '\e'
<unknown>:6: DeprecationWarning: invalid escape sequence '\i'
<unknown>:5: DeprecationWarning: invalid escape sequence '\d'
<unknown>:5: DeprecationWarning: invalid escape sequence '\k'
<unknown>:6: DeprecationWarning: invalid escape sequence '\o'
<unknown>:5: DeprecationWarning: invalid escape sequence '\c'
<unknown>:4: DeprecationWarning: invalid escape sequence '\D'
<unknown>:4: DeprecationWarning: invalid escape sequence '\#'
<unknown>:6: DeprecationWarning: invalid escape sequence '\G'
<unknown>:6: DeprecationWarning: invalid escape sequence '\p'
<unknown>:6: DeprecationWarning: invalid escape sequence '\P'
<unknown>:3: DeprecationWarning: invalid escape sequence '\|'
<unknown>:3: DeprecationWarning: invalid escape sequence '\s'
<unknown>:3: DeprecationWarning: invalid escape sequence '\ '
<unknown>:3: DeprecationWarning: invalid escape sequence '\p'
<unknown>:3: DeprecationWarning: invalid escape sequence '\l'
<unknown>:3: DeprecationWarning: invalid escape sequence '\|'
<unknown>:4: DeprecationWarning: invalid escape sequence '\z'
<unknown>:4: DeprecationWarning: invalid escape sequence '\g'
<unknown>:4: DeprecationWarning: invalid escape sequence '\s'
<unknown>:5: DeprecationWarning: invalid escape sequence '\D'
[01:50:48] WARNING  Unexpectedly had no exception name raised  syntaxtree.py:223
                    and no exception in context.                                
<unknown>:6: DeprecationWarning: invalid escape sequence '\i'
<unknown>:4: DeprecationWarning: invalid escape sequence '\m'
<unknown>:4: DeprecationWarning: invalid escape sequence '\d'
<unknown>:5: DeprecationWarning: invalid escape sequence '\c'
<unknown>:4: DeprecationWarning: invalid escape sequence '\{'
<unknown>:6: DeprecationWarning: invalid escape sequence '\i'
<unknown>:3: DeprecationWarning: invalid escape sequence '\ '
<unknown>:3: DeprecationWarning: invalid escape sequence '\.'
<unknown>:3: DeprecationWarning: invalid escape sequence '\*'
[01:50:50] INFO     C-extension module not found: _fblas          module.py:1372
           INFO     C-extension module not found: _flapack        module.py:1372
<unknown>:3: DeprecationWarning: invalid escape sequence '\|'
[01:50:51] INFO     C-extension module not found: _interpolative  module.py:1372
<unknown>:3: DeprecationWarning: invalid escape sequence '\_'
<unknown>:3: DeprecationWarning: invalid escape sequence '\p'
<unknown>:3: DeprecationWarning: invalid escape sequence '\s'
<unknown>:3: DeprecationWarning: invalid escape sequence '\i'
<unknown>:3: DeprecationWarning: invalid escape sequence '\e'
<unknown>:3: DeprecationWarning: invalid escape sequence '\G'
<unknown>:3: DeprecationWarning: invalid escape sequence '\i'
<unknown>:3: DeprecationWarning: invalid escape sequence '\L'
<unknown>:3: DeprecationWarning: invalid escape sequence '\m'
<unknown>:3: DeprecationWarning: invalid escape sequence '\p'
<unknown>:3: DeprecationWarning: invalid escape sequence '\z'
<unknown>:3: DeprecationWarning: invalid escape sequence '\s'
<unknown>:3: DeprecationWarning: invalid escape sequence '\l'
<unknown>:3: DeprecationWarning: invalid escape sequence '\c'
<unknown>:3: DeprecationWarning: invalid escape sequence '\g'
<unknown>:3: DeprecationWarning: invalid escape sequence '\*'
<unknown>:3: DeprecationWarning: invalid escape sequence '\l'
<unknown>:4: DeprecationWarning: invalid escape sequence '\*'
<unknown>:3: DeprecationWarning: invalid escape sequence '\m'
<unknown>:3: DeprecationWarning: invalid escape sequence '\o'
<unknown>:3: DeprecationWarning: invalid escape sequence '\D'
<unknown>:3: DeprecationWarning: invalid escape sequence '\m'
<unknown>:3: DeprecationWarning: invalid escape sequence '\i'
<unknown>:3: DeprecationWarning: invalid escape sequence '\ '
<unknown>:3: DeprecationWarning: invalid escape sequence '\i'
<unknown>:3: DeprecationWarning: invalid escape sequence '\p'
<unknown>:3: DeprecationWarning: invalid escape sequence '\e'
[01:50:57] WARNING  Unexpectedly had no exception name raised  syntaxtree.py:223
                    and no exception in context.                                
<unknown>:3: DeprecationWarning: invalid escape sequence '\p'
<unknown>:3: DeprecationWarning: invalid escape sequence '\i'
<unknown>:3: DeprecationWarning: invalid escape sequence '\s'
<unknown>:4: DeprecationWarning: invalid escape sequence '\s'
<unknown>:3: DeprecationWarning: invalid escape sequence '\p'
<unknown>:4: DeprecationWarning: invalid escape sequence '\l'
<unknown>:4: DeprecationWarning: invalid escape sequence '\c'
<unknown>:4: DeprecationWarning: invalid escape sequence '\e'
<unknown>:4: DeprecationWarning: invalid escape sequence '\s'
<unknown>:4: DeprecationWarning: invalid escape sequence '\l'
<unknown>:3: DeprecationWarning: invalid escape sequence '\o'
<unknown>:3: DeprecationWarning: invalid escape sequence '\m'
<unknown>:3: DeprecationWarning: invalid escape sequence '\h'
<unknown>:3: DeprecationWarning: invalid escape sequence '\i'
<unknown>:3: DeprecationWarning: invalid escape sequence '\,'
<unknown>:4: DeprecationWarning: invalid escape sequence '\k'
<unknown>:4: DeprecationWarning: invalid escape sequence '\m'
<unknown>:3: DeprecationWarning: invalid escape sequence '\l'
<unknown>:3: DeprecationWarning: invalid escape sequence '\h'
<unknown>:4: DeprecationWarning: invalid escape sequence '\c'
<unknown>:3: DeprecationWarning: invalid escape sequence '\,'
<unknown>:3: DeprecationWarning: invalid escape sequence '\g'
<unknown>:3: DeprecationWarning: invalid escape sequence '\s'
<unknown>:3: DeprecationWarning: invalid escape sequence '\ '
[01:51:00] INFO     C-extension module not found: uarray          module.py:1372
           INFO     C-extension module not found: uarray          module.py:1372
           INFO     C-extension module not found: uarray          module.py:1372
           INFO     C-extension module not found: uarray          module.py:1372
           INFO     C-extension module not found: uarray          module.py:1372
<unknown>:3: DeprecationWarning: invalid escape sequence '\s'
<unknown>:3: DeprecationWarning: invalid escape sequence '\|'
<unknown>:3: DeprecationWarning: invalid escape sequence '\c'
<unknown>:3: DeprecationWarning: invalid escape sequence '\m'
[01:51:01] INFO     C-extension module not found: _lbfgsb         module.py:1372
           INFO     C-extension module not found: _cobyla         module.py:1372
           INFO     C-extension module not found: _slsqp          module.py:1372
           INFO     C-extension module not found: _minpack        module.py:1372
[01:51:02] INFO     C-extension module not found: _odepack        module.py:1372
           INFO     C-extension module not found: _quadpack       module.py:1372
           INFO     C-extension module not found: _vode           module.py:1372
           INFO     C-extension module not found: _dop            module.py:1372
           INFO     C-extension module not found: _lsoda          module.py:1372
           INFO     C-extension module not found: _fitpack        module.py:1372
           INFO     C-extension module not found: _dfitpack       module.py:1372
           WARNING  Unexpectedly had no exception name raised  syntaxtree.py:223
                    and no exception in context.                                
           WARNING  Unexpectedly had no exception name raised  syntaxtree.py:223
                    and no exception in context.                                
<unknown>:3: DeprecationWarning: invalid escape sequence '\s'
<unknown>:3: DeprecationWarning: invalid escape sequence '\c'
           INFO     C-extension module not found: _mvn            module.py:1372
<unknown>:3: DeprecationWarning: invalid escape sequence '\l'
<unknown>:3: DeprecationWarning: invalid escape sequence '\m'
[01:51:04] INFO     C-extension module not found: _spropack       module.py:1372
           INFO     C-extension module not found: _dpropack       module.py:1372
           INFO     C-extension module not found: _cpropack       module.py:1372
           INFO     C-extension module not found: _zpropack       module.py:1372
           ERROR    While getting the types of exceptions in   syntaxtree.py:341
                    the handler, expected to find an ast.Name,                  
                    ast.Tuple, or ast.Attribute,but got                         
                    <ast.Call object at 0x7fce8e00e980>                         
           ERROR    While getting the types of exceptions in   syntaxtree.py:341
                    the handler, expected to find an ast.Name,                  
                    ast.Tuple, or ast.Attribute,but got                         
                    <ast.Call object at 0x7fce8e00ded0>                         
           INFO     C-extension module not found: _arpack         module.py:1372
<unknown>:3: DeprecationWarning: invalid escape sequence '\ '
[01:51:06] INFO     Analyzed project to create test cluster       module.py:1326
           INFO     Modules:     948                              module.py:1327
           INFO     Functions:  5458                              module.py:1328
           INFO     Classes:    1621                              module.py:1329
           INFO     Using seed 1733104240092726746              generator.py:202
           INFO     Using strategy:            generationalgorithmfactory.py:302
                    Algorithm.DYNAMOSA                                          
           INFO     Instantiated 26 fitness    generationalgorithmfactory.py:393
                    functions                                                   
           INFO     Using CoverageArchive      generationalgorithmfactory.py:346
           INFO     Using selection function:  generationalgorithmfactory.py:321
                    Selection.TOURNAMENT_SELEC                                  
                    TION                                                        
           INFO     Using crossover function:  generationalgorithmfactory.py:334
                    SinglePointRelativeCrossOv                                  
                    er                                                          
           INFO     Using ranking function:    generationalgorithmfactory.py:354
                    RankBasedPreferenceSorting                                  
           INFO     Start generating test cases                 generator.py:526
[01:51:16] INFO     Initial Population, Coverage: 0.153846  searchobserver.py:77
[01:51:24] INFO     Iteration:       3, Coverage: 0.307692  searchobserver.py:83
[01:51:25] INFO     Iteration:       4, Coverage: 0.307692  searchobserver.py:83
[01:51:27] INFO     Iteration:       5, Coverage: 0.384615  searchobserver.py:83
[01:51:29] INFO     Iteration:       6, Coverage: 0.423077  searchobserver.py:83
[01:51:30] INFO     Iteration:       7, Coverage: 0.423077  searchobserver.py:83
[01:51:32] INFO     Iteration:       8, Coverage: 0.423077  searchobserver.py:83
[01:51:34] INFO     Iteration:       9, Coverage: 0.423077  searchobserver.py:83
[01:51:36] INFO     Iteration:      10, Coverage: 0.423077  searchobserver.py:83
[01:51:37] INFO     Iteration:      11, Coverage: 0.461538  searchobserver.py:83
[01:51:38] INFO     Iteration:      12, Coverage: 0.461538  searchobserver.py:83
[01:51:40] INFO     Iteration:      13, Coverage: 0.461538  searchobserver.py:83
[01:51:42] INFO     Iteration:      14, Coverage: 0.461538  searchobserver.py:83
[01:51:43] INFO     Iteration:      15, Coverage: 0.461538  searchobserver.py:83
[01:51:45] INFO     Iteration:      16, Coverage: 0.461538  searchobserver.py:83
[01:51:46] INFO     Iteration:      17, Coverage: 0.461538  searchobserver.py:83
[01:51:48] INFO     Iteration:      18, Coverage: 0.461538  searchobserver.py:83
[01:51:50] INFO     Iteration:      19, Coverage: 0.461538  searchobserver.py:83
[01:51:51] INFO     Iteration:      20, Coverage: 0.461538  searchobserver.py:83
[01:51:52] INFO     Iteration:      21, Coverage: 0.461538  searchobserver.py:83
[01:51:55] INFO     Iteration:      22, Coverage: 0.461538  searchobserver.py:83
[01:51:57] INFO     Iteration:      23, Coverage: 0.461538  searchobserver.py:83
[01:51:59] INFO     Iteration:      24, Coverage: 0.461538  searchobserver.py:83
[01:52:00] INFO     Iteration:      25, Coverage: 0.461538  searchobserver.py:83
[01:52:01] INFO     Iteration:      26, Coverage: 0.461538  searchobserver.py:83
[01:52:04] INFO     Iteration:      27, Coverage: 0.461538  searchobserver.py:83
[01:52:05] INFO     Iteration:      28, Coverage: 0.461538  searchobserver.py:83
[01:52:07] INFO     Iteration:      29, Coverage: 0.461538  searchobserver.py:83
[01:52:10] INFO     Iteration:      30, Coverage: 0.461538  searchobserver.py:83
[01:52:11] INFO     Iteration:      31, Coverage: 0.461538  searchobserver.py:83
[01:52:13] INFO     Iteration:      32, Coverage: 0.461538  searchobserver.py:83
[01:52:16] INFO     Iteration:      33, Coverage: 0.461538  searchobserver.py:83
[01:52:17] INFO     Iteration:      34, Coverage: 0.461538  searchobserver.py:83
[01:52:18] INFO     Iteration:      35, Coverage: 0.461538  searchobserver.py:83
[01:52:20] INFO     Iteration:      36, Coverage: 0.461538  searchobserver.py:83
[01:52:22] INFO     Iteration:      37, Coverage: 0.461538  searchobserver.py:83
[01:52:23] INFO     Iteration:      38, Coverage: 0.461538  searchobserver.py:83
[01:52:25] INFO     Iteration:      39, Coverage: 0.461538  searchobserver.py:83
[01:52:26] INFO     Iteration:      40, Coverage: 0.461538  searchobserver.py:83
[01:52:28] INFO     Iteration:      41, Coverage: 0.461538  searchobserver.py:83
[01:52:30] INFO     Iteration:      42, Coverage: 0.461538  searchobserver.py:83
[01:52:31] INFO     Iteration:      43, Coverage: 0.461538  searchobserver.py:83
[01:52:33] INFO     Iteration:      44, Coverage: 0.461538  searchobserver.py:83
[01:52:35] INFO     Iteration:      45, Coverage: 0.461538  searchobserver.py:83
[01:52:37] INFO     Iteration:      46, Coverage: 0.461538  searchobserver.py:83
[01:52:39] INFO     Iteration:      47, Coverage: 0.538462  searchobserver.py:83
[01:52:41] INFO     Iteration:      48, Coverage: 0.538462  searchobserver.py:83
[01:52:42] INFO     Iteration:      49, Coverage: 0.538462  searchobserver.py:83
[01:52:43] INFO     Iteration:      50, Coverage: 0.538462  searchobserver.py:83
[01:52:44] INFO     Iteration:      51, Coverage: 0.538462  searchobserver.py:83
[01:52:45] INFO     Iteration:      52, Coverage: 0.538462  searchobserver.py:83
[01:52:48] INFO     Iteration:      53, Coverage: 0.538462  searchobserver.py:83
[01:52:50] INFO     Iteration:      54, Coverage: 0.538462  searchobserver.py:83
[01:52:51] INFO     Iteration:      55, Coverage: 0.538462  searchobserver.py:83
           ERROR    Exception in Thread: <Thread(Thread-2235   execution.py:2164
                    (_execute_test_case), started daemon                        
                    140525043844864)>                                           
                    ╭── Traceback (most recent call last) ───╮                  
                    │ /opt/conda/envs/redistricting/lib/pyth │                  
                    │ on3.10/threading.py:1016 in            │                  
                    │ _bootstrap_inner                       │                  
                    │                                        │                  
                    │   1013 │   │   │   │   _sys.setprofile │                  
                    │   1014 │   │   │                       │                  
                    │   1015 │   │   │   try:                │                  
                    │ ❱ 1016 │   │   │   │   self.run()      │                  
                    │   1017 │   │   │   except:             │                  
                    │   1018 │   │   │   │   self._invoke_ex │                  
                    │   1019 │   │   finally:                │                  
                    │                                        │                  
                    │ /opt/conda/envs/redistricting/lib/pyth │                  
                    │ on3.10/threading.py:953 in run         │                  
                    │                                        │                  
                    │    950 │   │   """                     │                  
                    │    951 │   │   try:                    │                  
                    │    952 │   │   │   if self._target is  │                  
                    │ ❱  953 │   │   │   │   self._target(*s │                  
                    │    954 │   │   finally:                │                  
                    │    955 │   │   │   # Avoid a refcycle  │                  
                    │    956 │   │   │   # an argument that  │                  
                    │                                        │                  
                    │ /opt/conda/envs/redistricting/lib/pyth │                  
                    │ on3.10/site-packages/pynguin/testcase/ │                  
                    │ execution.py:2268 in                   │                  
                    │ _execute_test_case                     │                  
                    │                                        │                  
                    │   2265 │   │   for idx, statement in e │                  
                    │   2266 │   │   │   ast_node = self._be │                  
                    │   2267 │   │   │   exception = self.ex │                  
                    │ ❱ 2268 │   │   │   self._after_stateme │                  
                    │   2269 │   │   │   if exception is not │                  
                    │   2270 │   │   │   │   result.report_n │                  
                    │   2271 │   │   │   │   break           │                  
                    │                                        │                  
                    │ /opt/conda/envs/redistricting/lib/pyth │                  
                    │ on3.10/site-packages/pynguin/testcase/ │                  
                    │ execution.py:2369 in                   │                  
                    │ _after_statement_execution             │                  
                    │                                        │                  
                    │   2366 │   │   # See comments in _befo │                  
                    │   2367 │   │   if self.tracer.current_ │                  
                    │   2368 │   │   │   # Kill this thread  │                  
                    │ ❱ 2369 │   │   │   raise RuntimeError( │                  
                    │   2370 │   │   │   │   "The current th │                  
                    │   2371 │   │   │   )                   │                  
                    │   2372                                 │                  
                    ╰────────────────────────────────────────╯                  
                    RuntimeError: The current thread shall not                  
                    be executed any more, thus I kill it.                       
           INFO     Iteration:      56, Coverage: 0.538462  searchobserver.py:83
[01:52:59] INFO     Iteration:      57, Coverage: 0.538462  searchobserver.py:83
[01:53:00] INFO     Iteration:      58, Coverage: 0.538462  searchobserver.py:83
[01:53:01] INFO     Iteration:      59, Coverage: 0.538462  searchobserver.py:83
[01:53:03] INFO     Iteration:      60, Coverage: 0.538462  searchobserver.py:83
[01:53:05] INFO     Iteration:      61, Coverage: 0.538462  searchobserver.py:83
[01:53:07] INFO     Iteration:      62, Coverage: 0.538462  searchobserver.py:83
[01:53:09] INFO     Iteration:      63, Coverage: 0.538462  searchobserver.py:83
[01:53:10] INFO     Iteration:      64, Coverage: 0.538462  searchobserver.py:83
[01:53:12] INFO     Iteration:      65, Coverage: 0.538462  searchobserver.py:83
[01:53:13] INFO     Iteration:      66, Coverage: 0.538462  searchobserver.py:83
[01:53:14] INFO     Iteration:      67, Coverage: 0.538462  searchobserver.py:83
[01:53:16] INFO     Iteration:      68, Coverage: 0.538462  searchobserver.py:83
[01:53:18] INFO     Iteration:      69, Coverage: 0.538462  searchobserver.py:83
[01:53:19] INFO     Iteration:      70, Coverage: 0.538462  searchobserver.py:83
[01:53:20] INFO     Iteration:      71, Coverage: 0.538462  searchobserver.py:83
[01:53:22] INFO     Iteration:      72, Coverage: 0.538462  searchobserver.py:83
[01:53:23] INFO     Iteration:      73, Coverage: 0.538462  searchobserver.py:83
[01:53:25] INFO     Iteration:      74, Coverage: 0.538462  searchobserver.py:83
[01:53:26] INFO     Iteration:      75, Coverage: 0.538462  searchobserver.py:83
[01:53:29] INFO     Iteration:      76, Coverage: 0.538462  searchobserver.py:83
[01:53:31] INFO     Iteration:      77, Coverage: 0.538462  searchobserver.py:83
[01:53:32] INFO     Iteration:      78, Coverage: 0.538462  searchobserver.py:83
[01:53:35] INFO     Iteration:      79, Coverage: 0.538462  searchobserver.py:83
[01:53:37] INFO     Iteration:      80, Coverage: 0.538462  searchobserver.py:83
[01:53:38] INFO     Iteration:      81, Coverage: 0.538462  searchobserver.py:83
[01:53:40] INFO     Iteration:      82, Coverage: 0.615385  searchobserver.py:83
[01:53:42] INFO     Iteration:      83, Coverage: 0.615385  searchobserver.py:83
[01:53:44] INFO     Iteration:      84, Coverage: 0.615385  searchobserver.py:83
[01:53:45] INFO     Iteration:      85, Coverage: 0.615385  searchobserver.py:83
[01:53:47] INFO     Iteration:      86, Coverage: 0.615385  searchobserver.py:83
[01:53:49] INFO     Iteration:      87, Coverage: 0.615385  searchobserver.py:83
[01:53:50] INFO     Iteration:      88, Coverage: 0.615385  searchobserver.py:83
[01:53:52] INFO     Iteration:      89, Coverage: 0.615385  searchobserver.py:83
[01:53:54] INFO     Iteration:      90, Coverage: 0.615385  searchobserver.py:83
[01:53:56] INFO     Iteration:      91, Coverage: 0.615385  searchobserver.py:83
[01:53:57] INFO     Iteration:      92, Coverage: 0.615385  searchobserver.py:83
[01:53:59] INFO     Iteration:      93, Coverage: 0.615385  searchobserver.py:83
[01:54:00] INFO     Iteration:      94, Coverage: 0.615385  searchobserver.py:83
[01:54:01] INFO     Iteration:      95, Coverage: 0.615385  searchobserver.py:83
[01:54:03] INFO     Iteration:      96, Coverage: 0.615385  searchobserver.py:83
[01:54:05] INFO     Iteration:      97, Coverage: 0.615385  searchobserver.py:83
[01:54:07] INFO     Iteration:      98, Coverage: 0.615385  searchobserver.py:83
           INFO     Iteration:      99, Coverage: 0.615385  searchobserver.py:83
           ERROR    Exception in Thread: <Thread(Thread-3970   execution.py:2164
                    (_execute_test_case), started daemon                        
                    140525043844864)>                                           
                    ╭── Traceback (most recent call last) ───╮                  
                    │ /opt/conda/envs/redistricting/lib/pyth │                  
                    │ on3.10/threading.py:1016 in            │                  
                    │ _bootstrap_inner                       │                  
                    │                                        │                  
                    │   1013 │   │   │   │   _sys.setprofile │                  
                    │   1014 │   │   │                       │                  
                    │   1015 │   │   │   try:                │                  
                    │ ❱ 1016 │   │   │   │   self.run()      │                  
                    │   1017 │   │   │   except:             │                  
                    │   1018 │   │   │   │   self._invoke_ex │                  
                    │   1019 │   │   finally:                │                  
                    │                                        │                  
                    │ /opt/conda/envs/redistricting/lib/pyth │                  
                    │ on3.10/threading.py:953 in run         │                  
                    │                                        │                  
                    │    950 │   │   """                     │                  
                    │    951 │   │   try:                    │                  
                    │    952 │   │   │   if self._target is  │                  
                    │ ❱  953 │   │   │   │   self._target(*s │                  
                    │    954 │   │   finally:                │                  
                    │    955 │   │   │   # Avoid a refcycle  │                  
                    │    956 │   │   │   # an argument that  │                  
                    │                                        │                  
                    │ /opt/conda/envs/redistricting/lib/pyth │                  
                    │ on3.10/site-packages/pynguin/testcase/ │                  
                    │ execution.py:2268 in                   │                  
                    │ _execute_test_case                     │                  
                    │                                        │                  
                    │   2265 │   │   for idx, statement in e │                  
                    │   2266 │   │   │   ast_node = self._be │                  
                    │   2267 │   │   │   exception = self.ex │                  
                    │ ❱ 2268 │   │   │   self._after_stateme │                  
                    │   2269 │   │   │   if exception is not │                  
                    │   2270 │   │   │   │   result.report_n │                  
                    │   2271 │   │   │   │   break           │                  
                    │                                        │                  
                    │ /opt/conda/envs/redistricting/lib/pyth │                  
                    │ on3.10/site-packages/pynguin/testcase/ │                  
                    │ execution.py:2369 in                   │                  
                    │ _after_statement_execution             │                  
                    │                                        │                  
                    │   2366 │   │   # See comments in _befo │                  
                    │   2367 │   │   if self.tracer.current_ │                  
                    │   2368 │   │   │   # Kill this thread  │                  
                    │ ❱ 2369 │   │   │   raise RuntimeError( │                  
                    │   2370 │   │   │   │   "The current th │                  
                    │   2371 │   │   │   )                   │                  
                    │   2372                                 │                  
                    ╰────────────────────────────────────────╯                  
                    RuntimeError: The current thread shall not                  
                    be executed any more, thus I kill it.                       
           INFO     Iteration:     100, Coverage: 0.615385  searchobserver.py:83
[01:54:18] INFO     Iteration:     101, Coverage: 0.615385  searchobserver.py:83
[01:54:20] INFO     Iteration:     102, Coverage: 0.615385  searchobserver.py:83
[01:54:21] INFO     Iteration:     103, Coverage: 0.615385  searchobserver.py:83
[01:54:23] INFO     Iteration:     104, Coverage: 0.615385  searchobserver.py:83
[01:54:30] INFO     Iteration:     105, Coverage: 0.615385  searchobserver.py:83
[01:54:31] INFO     Iteration:     106, Coverage: 0.615385  searchobserver.py:83
[01:54:33] INFO     Iteration:     107, Coverage: 0.615385  searchobserver.py:83
[01:54:36] INFO     Iteration:     108, Coverage: 0.615385  searchobserver.py:83
[01:54:39] INFO     Iteration:     109, Coverage: 0.615385  searchobserver.py:83
[01:54:40] INFO     Iteration:     110, Coverage: 0.615385  searchobserver.py:83
[01:54:41] INFO     Iteration:     111, Coverage: 0.615385  searchobserver.py:83
[01:54:42] INFO     Iteration:     112, Coverage: 0.615385  searchobserver.py:83
[01:54:45] INFO     Iteration:     113, Coverage: 0.615385  searchobserver.py:83
[01:54:47] INFO     Iteration:     114, Coverage: 0.615385  searchobserver.py:83
[01:54:48] INFO     Iteration:     115, Coverage: 0.615385  searchobserver.py:83
[01:54:50] INFO     Iteration:     116, Coverage: 0.615385  searchobserver.py:83
[01:54:51] INFO     Iteration:     117, Coverage: 0.615385  searchobserver.py:83
[01:54:53] INFO     Iteration:     118, Coverage: 0.615385  searchobserver.py:83
[01:54:55] INFO     Iteration:     119, Coverage: 0.615385  searchobserver.py:83
[01:54:56] INFO     Iteration:     120, Coverage: 0.615385  searchobserver.py:83
[01:54:59] INFO     Iteration:     121, Coverage: 0.615385  searchobserver.py:83
[01:55:01] INFO     Iteration:     122, Coverage: 0.615385  searchobserver.py:83
[01:55:04] INFO     Iteration:     123, Coverage: 0.615385  searchobserver.py:83
[01:55:05] INFO     Iteration:     124, Coverage: 0.615385  searchobserver.py:83
[01:55:07] INFO     Iteration:     125, Coverage: 0.615385  searchobserver.py:83
[01:55:09] INFO     Iteration:     126, Coverage: 0.615385  searchobserver.py:83
[01:55:11] INFO     Iteration:     127, Coverage: 0.615385  searchobserver.py:83
[01:55:13] INFO     Iteration:     128, Coverage: 0.615385  searchobserver.py:83
[01:55:15] INFO     Iteration:     129, Coverage: 0.615385  searchobserver.py:83
[01:55:17] INFO     Iteration:     130, Coverage: 0.615385  searchobserver.py:83
[01:55:18] INFO     Iteration:     131, Coverage: 0.615385  searchobserver.py:83
[01:55:19] INFO     Iteration:     132, Coverage: 0.615385  searchobserver.py:83
[01:55:20] INFO     Iteration:     133, Coverage: 0.615385  searchobserver.py:83
[01:55:22] INFO     Iteration:     134, Coverage: 0.615385  searchobserver.py:83
[01:55:24] INFO     Iteration:     135, Coverage: 0.615385  searchobserver.py:83
[01:55:25] INFO     Iteration:     136, Coverage: 0.615385  searchobserver.py:83
[01:55:27] INFO     Iteration:     137, Coverage: 0.615385  searchobserver.py:83
[01:55:28] INFO     Iteration:     138, Coverage: 0.615385  searchobserver.py:83
[01:55:29] INFO     Iteration:     139, Coverage: 0.615385  searchobserver.py:83
[01:55:32] INFO     Iteration:     140, Coverage: 0.615385  searchobserver.py:83
[01:55:34] INFO     Iteration:     141, Coverage: 0.615385  searchobserver.py:83
[01:55:35] INFO     Iteration:     142, Coverage: 0.615385  searchobserver.py:83
[01:55:36] INFO     Iteration:     143, Coverage: 0.615385  searchobserver.py:83
[01:55:38] INFO     Iteration:     144, Coverage: 0.615385  searchobserver.py:83
[01:55:39] INFO     Iteration:     145, Coverage: 0.615385  searchobserver.py:83
[01:55:41] INFO     Iteration:     146, Coverage: 0.615385  searchobserver.py:83
[01:55:44] INFO     Iteration:     147, Coverage: 0.615385  searchobserver.py:83
[01:55:47] INFO     Iteration:     148, Coverage: 0.615385  searchobserver.py:83
[01:55:48] INFO     Iteration:     149, Coverage: 0.615385  searchobserver.py:83
[01:55:50] INFO     Iteration:     150, Coverage: 0.615385  searchobserver.py:83
[01:55:51] INFO     Iteration:     151, Coverage: 0.615385  searchobserver.py:83
[01:55:53] INFO     Iteration:     152, Coverage: 0.615385  searchobserver.py:83
[01:55:55] INFO     Iteration:     153, Coverage: 0.615385  searchobserver.py:83
[01:55:57] INFO     Iteration:     154, Coverage: 0.615385  searchobserver.py:83
[01:55:59] INFO     Iteration:     155, Coverage: 0.615385  searchobserver.py:83
[01:56:00] INFO     Iteration:     156, Coverage: 0.615385  searchobserver.py:83
[01:56:02] INFO     Iteration:     157, Coverage: 0.615385  searchobserver.py:83
[01:56:04] INFO     Iteration:     158, Coverage: 0.615385  searchobserver.py:83
[01:56:05] INFO     Iteration:     159, Coverage: 0.615385  searchobserver.py:83
[01:56:08] INFO     Iteration:     160, Coverage: 0.615385  searchobserver.py:83
           INFO     Stopping condition reached                  generator.py:531
           INFO     Used search time: 301/300                   generator.py:533
           INFO     Stop generating test cases                  generator.py:534
           INFO     Start generating assertions                 generator.py:664
           INFO     Setup mutation generator                    generator.py:637
           INFO     Import module spec_kmeans                   generator.py:640
           INFO     Build AST for spec_kmeans                   generator.py:643
           INFO     Mutate module spec_kmeans                   generator.py:648
           INFO     Generated 39 mutants                        generator.py:657
           INFO     Running tests on mutant   1/39     assertiongenerator.py:346
           INFO     Running tests on mutant   2/39     assertiongenerator.py:346
           INFO     Running tests on mutant   3/39     assertiongenerator.py:346
           INFO     Running tests on mutant   4/39     assertiongenerator.py:346
           INFO     Running tests on mutant   5/39     assertiongenerator.py:346
           INFO     Running tests on mutant   6/39     assertiongenerator.py:346
           INFO     Running tests on mutant   7/39     assertiongenerator.py:346
           INFO     Running tests on mutant   8/39     assertiongenerator.py:346
           INFO     Running tests on mutant   9/39     assertiongenerator.py:346
[01:56:09] INFO     Running tests on mutant  10/39     assertiongenerator.py:346
           INFO     Running tests on mutant  11/39     assertiongenerator.py:346
           INFO     Running tests on mutant  12/39     assertiongenerator.py:346
           INFO     Running tests on mutant  13/39     assertiongenerator.py:346
           INFO     Running tests on mutant  14/39     assertiongenerator.py:346
           INFO     Running tests on mutant  15/39     assertiongenerator.py:346
           INFO     Running tests on mutant  16/39     assertiongenerator.py:346
           INFO     Running tests on mutant  17/39     assertiongenerator.py:346
           INFO     Running tests on mutant  18/39     assertiongenerator.py:346
           INFO     Running tests on mutant  19/39     assertiongenerator.py:346
           INFO     Running tests on mutant  20/39     assertiongenerator.py:346
           INFO     Running tests on mutant  21/39     assertiongenerator.py:346
           INFO     Running tests on mutant  22/39     assertiongenerator.py:346
           INFO     Running tests on mutant  23/39     assertiongenerator.py:346
           INFO     Running tests on mutant  24/39     assertiongenerator.py:346
           INFO     Running tests on mutant  25/39     assertiongenerator.py:346
           INFO     Running tests on mutant  26/39     assertiongenerator.py:346
[01:56:10] INFO     Running tests on mutant  27/39     assertiongenerator.py:346
           INFO     Running tests on mutant  28/39     assertiongenerator.py:346
           INFO     Running tests on mutant  29/39     assertiongenerator.py:346
           INFO     Running tests on mutant  30/39     assertiongenerator.py:346
           INFO     Running tests on mutant  31/39     assertiongenerator.py:346
           INFO     Running tests on mutant  32/39     assertiongenerator.py:346
           INFO     Running tests on mutant  33/39     assertiongenerator.py:346
           INFO     Running tests on mutant  34/39     assertiongenerator.py:346
           INFO     Running tests on mutant  35/39     assertiongenerator.py:346
           INFO     Running tests on mutant  36/39     assertiongenerator.py:346
           INFO     Running tests on mutant  37/39     assertiongenerator.py:346
           INFO     Running tests on mutant  38/39     assertiongenerator.py:346
           INFO     Running tests on mutant  39/39     assertiongenerator.py:346
           INFO     Mutant 0 killed by Test(s): 0, 1,  assertiongenerator.py:423
                    2, 3, 4, 5, 6, 8, 9, 10                                     
           INFO     Mutant 1 killed by Test(s): 0, 1,  assertiongenerator.py:423
                    2, 3, 4, 5, 6, 8, 9, 10                                     
           INFO     Mutant 2 killed by Test(s): 0, 1,  assertiongenerator.py:423
                    2, 3, 4, 5, 6, 8, 9, 10                                     
           INFO     Mutant 3 killed by Test(s): 0, 1,  assertiongenerator.py:423
                    2, 3, 4, 5, 6, 8, 9, 10                                     
           INFO     Mutant 4 killed by Test(s): 0, 1,  assertiongenerator.py:423
                    2, 3, 4, 5, 6, 8, 9, 10                                     
           INFO     Mutant 5 killed by Test(s): 0, 1,  assertiongenerator.py:423
                    2, 3, 4, 5, 6, 8, 9, 10                                     
           INFO     Mutant 6 killed by Test(s): 0, 1,  assertiongenerator.py:423
                    2, 3, 4, 5, 6, 8, 9, 10                                     
           INFO     Mutant 7 killed by Test(s): 0, 1,  assertiongenerator.py:423
                    2, 3, 4, 5, 6, 8, 9, 10                                     
           INFO     Mutant 8 killed by Test(s): 0, 1,  assertiongenerator.py:423
                    2, 3, 4, 5, 6, 8, 9, 10                                     
           INFO     Mutant 9 killed by Test(s): 0, 1,  assertiongenerator.py:423
                    2, 3, 4, 5, 6, 8, 9, 10                                     
           INFO     Mutant 10 killed by Test(s): 0, 1, assertiongenerator.py:423
                    2, 3, 4, 5, 6, 8, 9, 10                                     
           INFO     Mutant 11 killed by Test(s): 0, 1, assertiongenerator.py:423
                    2, 3, 4, 5, 6, 8, 9, 10                                     
           INFO     Mutant 12 killed by Test(s): 0, 1, assertiongenerator.py:423
                    2, 3, 4, 5, 6, 8, 9, 10                                     
           INFO     Mutant 13 killed by Test(s): 0, 1, assertiongenerator.py:423
                    2, 3, 4, 5, 6, 8, 9, 10                                     
           INFO     Mutant 14 killed by Test(s): 0, 1, assertiongenerator.py:423
                    2, 3, 4, 5, 6, 8, 9, 10                                     
           INFO     Mutant 15 killed by Test(s): 0, 1, assertiongenerator.py:423
                    2, 3, 4, 5, 6, 8, 9, 10                                     
           INFO     Mutant 16 killed by Test(s): 0, 1, assertiongenerator.py:423
                    2, 3, 4, 5, 6, 8, 9, 10                                     
           INFO     Mutant 17 killed by Test(s): 0, 1, assertiongenerator.py:423
                    2, 3, 4, 5, 6, 8, 9, 10                                     
           INFO     Mutant 18 killed by Test(s): 0, 1, assertiongenerator.py:423
                    2, 3, 4, 5, 6, 8, 9, 10                                     
           INFO     Mutant 19 killed by Test(s): 0, 1, assertiongenerator.py:423
                    2, 3, 4, 5, 6, 8, 9, 10                                     
           INFO     Mutant 20 killed by Test(s): 0, 1, assertiongenerator.py:423
                    2, 3, 4, 5, 6, 8, 9, 10                                     
           INFO     Mutant 21 killed by Test(s): 0, 1, assertiongenerator.py:423
                    2, 3, 4, 5, 6, 8, 9, 10                                     
           INFO     Mutant 22 killed by Test(s): 0, 1, assertiongenerator.py:423
                    2, 3, 4, 5, 6, 8, 9, 10                                     
           INFO     Mutant 23 killed by Test(s): 0, 1, assertiongenerator.py:423
                    2, 3, 4, 5, 6, 8, 9, 10                                     
           INFO     Mutant 24 killed by Test(s): 0, 1, assertiongenerator.py:423
                    2, 3, 4, 5, 6, 8, 9, 10                                     
           INFO     Mutant 25 killed by Test(s): 0, 1, assertiongenerator.py:423
                    2, 3, 4, 5, 6, 7, 8, 9, 10                                  
           INFO     Mutant 26 killed by Test(s): 0, 1, assertiongenerator.py:423
                    2, 3, 4, 5, 6, 8, 9, 10                                     
           INFO     Mutant 27 killed by Test(s): 0, 1, assertiongenerator.py:423
                    2, 3, 4, 5, 6, 8, 9, 10                                     
           INFO     Mutant 28 killed by Test(s): 0, 1, assertiongenerator.py:423
                    2, 3, 4, 5, 6, 8, 9, 10                                     
           INFO     Mutant 29 killed by Test(s): 0, 1, assertiongenerator.py:423
                    2, 3, 4, 5, 6, 8, 9, 10                                     
           INFO     Mutant 30 killed by Test(s): 0, 1, assertiongenerator.py:423
                    2, 3, 4, 5, 6, 8, 9, 10                                     
           INFO     Mutant 31 killed by Test(s): 0, 1, assertiongenerator.py:423
                    2, 3, 4, 5, 6, 8, 9, 10                                     
           INFO     Mutant 32 killed by Test(s): 0, 1, assertiongenerator.py:423
                    2, 3, 4, 5, 6, 8, 9, 10                                     
           INFO     Mutant 33 killed by Test(s): 0, 1, assertiongenerator.py:423
                    2, 3, 4, 5, 6, 8, 9, 10                                     
           INFO     Mutant 34 killed by Test(s): 0, 1, assertiongenerator.py:423
                    2, 3, 4, 5, 6, 7, 8, 9, 10                                  
           INFO     Mutant 35 killed by Test(s): 0, 1, assertiongenerator.py:423
                    2, 3, 4, 5, 6, 8, 9, 10                                     
           INFO     Mutant 36 killed by Test(s): 0, 1, assertiongenerator.py:423
                    2, 3, 4, 5, 6, 8, 9, 10                                     
           INFO     Mutant 37 killed by Test(s): 0, 1, assertiongenerator.py:423
                    2, 3, 4, 5, 6, 8, 9, 10                                     
           INFO     Mutant 38 killed by Test(s): 0, 1, assertiongenerator.py:423
                    2, 3, 4, 5, 6, 8, 9, 10                                     
           INFO     Number of Surviving Mutant(s): 0   assertiongenerator.py:435
                    (Mutants: )                                                 
           INFO     Calculating resulting FinalBranchCoverage   generator.py:447
           INFO     Written 11 test cases to                    generator.py:774
                    /home/user/out_DYNAMOSA/suite_0/test_spec_k                 
                    means.py                                                    
           INFO     Writing statistics                         statistics.py:370
           INFO     Stop Pynguin Test Generation…               generator.py:119
sys:1: ResourceWarning: unclosed file <_io.TextIOWrapper name='/dev/null' mode='w' encoding='UTF-8'>
ResourceWarning: Enable tracemalloc to get the object allocation traceback
